dialog/fileitem.go

```diff
@@ -55,6 +64,31 @@ func (i *fileDialogItem) setLocation(l fyne.URI, dir, up bool) {
 
 	i.Refresh()
 }
+func (i *fileDialogItem) tapped() {
+	if i.choose != nil {
+		i.choose(i.id)
+	}
+	now := time.Now()
+	if !i.dir && now.Sub(i.lastClick) < fyne.CurrentApp().Driver().DoubleTapDelay() && i.open != nil {
+		// It is a double click, so we ask the dialog to open
+		i.open()
+	}
+	i.lastClick = now
+}
+
+func (i *fileDialogItem) TouchDown(*mobile.TouchEvent) {}
```

Comment:
I think you can still use Tappable instead of Mouseable/Touchable. If your object is Tappable and not DoubleTappable then there will be no delay and you can just do the same logic you already have here to treat it as a single tap to select or a double tap to open.

Reply:
Yes. It is a more elegant way. I have switched to that.